openedx/core/djangoapps/credit/api.py

```diff
@@ -111,6 +111,93 @@ def get_credit_requirements(course_key, namespace=None):
     ]
 
 
+def get_credit_requirement_status(course_key, username):
+    """ Retrieve the user's status for each credit requirement in the course.
+
+    Args:
+        course_key (CourseKey): The identifier for course
+        username (str): The identifier of the user
+
+    Example:
+        >>> get_credit_requirement_status("course-v1-edX-DemoX-1T2015", "john")
+
+                [
+                    {
+                        "namespace": "verification",
+                        "name": "verification",
+                        "criteria": {},
+                        "status": "satisfied",
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "midterm",
+                        "criteria": {},
+                        "status": "Not satisfied",
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "final",
+                        "criteria": {},
+                        "status": "error",
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "grade",
+                        "criteria": {"min_grade": 0.8},
+                        "status": None,
+                    },
+                ]
+
+    Returns:
+        list of requirement statuses
+    """
+    requirements = CreditRequirement.get_course_requirements(course_key)
+    requirement_list = [requirement.id for requirement in requirements]
+    requirement_statuses = CreditRequirementStatus.get_statuses(requirement_list, username)
+    statuses = []
+    for requirement in requirements:
+        status = None
+        status_date = None
+        for requirement_status in requirement_statuses:
+            if requirement_status.requirement == requirement:
+                status = requirement_status.status
+                status_date = requirement_status.modified.strftime('%m/%d/%Y')
+                break
+        statuses.append({
+            "namespace": requirement.namespace,
+            "name": requirement.name,
+            "criteria": requirement.criteria,
+            "status": status,
+            "status_date": status_date,
+        })
+    return statuses
+
+
+def is_course_credit_eligible(username, course_key):
```

Comment:
This name doesn't describe this method. `is_user_eligible_for_credit` would be better.